frontend/test/specs/user/searchInstitutionControllerSpec.js

```diff
@@ -0,0 +1,134 @@
+    };
+
+    const otherInstitution = {
+        name: 'other institution',
+        key: 'other-inst-key',
+        id: 'other-inst-key'
+    };
+
+    const user = {
+        name: 'User',
+        key: 'user-key',
+        email: 'user@email',
+        state: 'active'
+    };
+
+    beforeEach(module('app'));
+
+    beforeEach(inject(function($componentController, $httpBackend, $rootScope, $state,
+         AuthService, InstitutionService, $q) {
+        
+        httpBackend = $httpBackend;
+        scope = $rootScope.$new();
+        state = $state;
+        institutionService = InstitutionService;
+        deferred = $q.defer();
+
+        AuthService.login(user);
+
+        searchInstCtrl = $componentController('searchInstitution', scope, {
+            onSelect: () => {},
+            onSearch: () => {}
+        });
+    }));
+
+    describe('UserInactiveController functions', function() {
+
+        describe('getInstIcon', function () {
+            it('should return the done icon', function() {
+                spyOn(searchInstCtrl, 'isInstSelected').and.returnValue(true);
+                expect(searchInstCtrl.getInstIcon(institution)).toEqual('done');
+            });
+
+            it('should return the account_balance icon', function() {
+                spyOn(searchInstCtrl, 'isInstSelected').and.returnValue(false);
+                expect(searchInstCtrl.getInstIcon(institution)).toEqual('account_balance');
+            });
+        });
+
+        describe('isInstSelected', function () {
+            it('should be true when the institution is selected and loaded', function() {
+                searchInstCtrl.selectedInst = new Institution(institution);
+                searchInstCtrl.isInstLoaded = true;
+                expect(searchInstCtrl.isInstSelected(institution)).toBeTruthy();
+            });
+
+            it('should be false when the institution is selected and not loaded', function() {
+                searchInstCtrl.selectedInst = new Institution(institution);
+                searchInstCtrl.isInstLoaded = false;
+                expect(searchInstCtrl.isInstSelected(institution)).toBeFalsy();
+            });
+
+            it('should be false when the institution is not the one selected and is loaded', function() {
+                searchInstCtrl.selectedInst = new Institution(otherInstitution);
+                searchInstCtrl.isInstLoaded = true;
+                expect(searchInstCtrl.isInstSelected(institution)).toBeFalsy();
+            });
+        })
+
+        describe('search()', function(){
+            beforeEach(function () {
+                spyOn(searchInstCtrl, 'onSearch');
+                spyOn(institutionService, 'searchInstitutions').and.returnValue(deferred.promise);
+            });
+            
+            it('Should return the institutions found', function(){
+                const institutions = [institution, otherInstitution];
+                deferred.resolve(institutions);
```

Comment:
Using the same promise and solving it twice can be a problem, right?

Reply:
The beforeEach executes before each test, so this is always reset before a new test.
More information: http://breazeal.com/blog/jasmineBefore.html